quinn-udp/src/windows.rs

```diff
@@ -434,21 +423,67 @@ static WSARECVMSG_PTR: Lazy<WinSock::LPFN_WSARECVMSG> = Lazy::new(|| {
 });
 
 static MAX_GSO_SEGMENTS: Lazy<usize> = Lazy::new(|| {
-    let socket = match std::net::UdpSocket::bind("[::]:0")
-        .or_else(|_| std::net::UdpSocket::bind("127.0.0.1:0"))
-    {
+    let socket = match std::net::UdpSocket::bind("127.0.0.1:0") {
         Ok(socket) => socket,
         Err(_) => return 1,
     };
-    const GSO_SIZE: c_uint = 1500;
-    match set_socket_option(
-        &socket,
-        WinSock::IPPROTO_UDP,
-        WinSock::UDP_SEND_MSG_SIZE,
-        GSO_SIZE,
-    ) {
-        // Empirically found on Windows 11 x64
-        Ok(()) => 512,
+
+    socket.set_nonblocking(true).ok();
+
+    // Test GSO support by actually trying to send with the UDP_SEND_MSG_SIZE control message.
+    // Using setsockopt() alone is not sufficient as it may succeed even when WSASendMsg()
+    // with the control message fails (e.g., due to network adapter/driver limitations).
```

Comment:
I would have assumed sending on localhost does not actually test driver support. Would you mind describing the bug you have been able to fix with this patch?